### DISCUSSION TITLE
Round an edge with ansys.geometry

### DISCUSSION THREAD

**Discussion by gze** [ansys/pyansys-geometry#1484] [Q&A] [answered]
Using ansys.geometry, I want to round an edge of a body, but I cannot find this feature. Is it available?
If not:
* Is it planned for the future?
* Can I run "Discovery IronPython" code in PyAnsys to add the round as a workaround?

**Comment by greschd** [comment]
Hi @gze 👋

While I don't know the answer to the question about rounding edges, you _can_ run Discovery scripts, with the [`run_discovery_script_file` method](https://geometry.docs.pyansys.com/version/stable/api/ansys/geometry/core/modeler/Modeler.html#Modeler.run_discovery_script_file).

I'll transfer the discussion to the PyAnsys Geometry repository for better visibility by those who might answer the question.. maybe @RobPasMue knows?

**Comment by RobPasMue** [comment, marked as answer]
Hi @gze!

Using PyAnsys Geometry you do have the option to use the native API to perform repair and prepare tools operations. Some of them however might not be available for you yet because we implemented native support for it in 25R1. See https://geometry.docs.pyansys.com/version/stable/api/ansys/geometry/core/tools/repair_tools/RepairTools.html#ansys.geometry.core.tools.repair_tools.RepairTools and https://geometry.docs.pyansys.com/version/stable/api/ansys/geometry/core/tools/prepare_tools/PrepareTools.html#ansys.geometry.core.tools.prepare_tools.PrepareTools

In any case, like @greschd mentions, if you have a DIscovery script already available you can use it as a backdoor to get your results! Also, if you do so, we would appreciate it if you let us know what API calls you are performing so that we can expose them natively.

**Comment by gze** [comment]
Creating the round using modeler.run_discovery_script_file() works.
I was not successful passing script_args arguments to argsDict (as mentionded in the docstring)

In my script using ansys.geomery.core i call:
```
modeler.run_discovery_script_file(
    file_path="round.py",
    script_args=dict(
        body_nr="-1",
        edge_nr="-1",
        radius="100",
    )
)
```

The round.py contains:
```
# Python Script, API Version = V241

# The script is called with the arguments
# run_discovery_script_file(..., script_args = {"length": "20"}, ...),
# the script will receive the dictionary argsDict with the key-value pair {"length": "20"}.

# argsDict = dict(body_nr="-1", edge_nr="-1", radius="100") #WORK IF I UNCOMMENT THIS LINE

body_nr = int(argsDict.get("body_nr"))
edge_nr = int(argsDict.get("edge_nr"))
radius = float(argsDict.get("radius", 10))

# Create 1 Round
selection = EdgeSelection.Create(GetRootPart().Bodies[body_nr].Edges[edge_nr])
options = ConstantRoundOptions()
result = ConstantRound.Execute(selection, MM(radius), options, None)
```

How do I get the script_args passed to the argsDict
(FYI I am currently on ansys-geometry-core 0.6.6 will upgrade my python to 3.10 to get the latest ansys-geometry-core version shortly)

**Comment by RobPasMue** [comment]
Hi @gze - have a look at the following screenshot

![image](https://github.com/user-attachments/assets/2f183c5c-2b20-4b30-83ad-bb6c847d5619)

In my case everything worked as expected as you can see in the output. Disco script in IronPython executed properly and the output was returned successfully to PyAnsys Geometry with the result object... Can you give it a try to access the dictionary without the ``get``? That threw me an error when trying it.

I am also using version 0.7.4 - maybe something got fixed in the meantime and I can't remember right now...

**Reply by gze** [reply]
Was indeed related to how the dict was accessed. 
Is working now. Thanks for the help.

**Reply by RobPasMue** [reply]
Glad it worked!

**Comment by JulianJM45** [comment]
Hi, is rounding an edge now natively included in pyansys.geometry in the latest version or do I still have to do this workaround ?

**Reply by RobPasMue** [reply]
Hi @JulianJM45 - as of 25R2 and onwards, users can perform chamfer and fillet operations on edges. The latter, I believe, would be what you are looking for:

```py
# Fillet radius
fillet_radius = 0.1

# Given an Edge object called "edge"
is_success = modeler.geometry_commands.fillet(selection=[edge], radius=fillet_radius)
print(f"Was fillet successful? {is_success}")
```

**Reply by JulianJM45** [reply]
thanks
